### PR TITLE
do not rely on .constructor.name in R.toString

### LINKED ISSUE #1393: do not rely on .constructor.name in R.toString
As mentioned in https://github.com/ramda/ramda/issues/1140#issuecomment-119953239, the implementation of `_toString` currently contains `x.constructor.name !== 'Object'`. Apparently not every environment provides a `name` property on each function. Instead we could use a regular expression such as `^function (\w*)` to extract a function's name from its string representation.

### PR DESCRIPTION
**Pull request by davidchambers**
Closes #1393

### CODE REVIEW

File: src/internal/_functionName.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports = function _functionName(f) {
+  // String(x => x) evaluates to "x => x", so the pattern may not match.
+  var match = String(f).match(/^function (\w*)/);
+  return match == null ? '' : match[1];
```
Comment (buzzdecafe) [inline src/internal/_functionName.js]: could you use `constructor.name` _if present_ and otherwise evaluate the stringified function?

Comment (davidchambers) [inline src/internal/_functionName.js]: Are you concerned about the performance implications?

Comment (buzzdecafe) [inline src/internal/_functionName.js]: not especially. but if we have it, why not use it?

Comment (davidchambers) [inline src/internal/_functionName.js]: Given a choice between one code path and two, I'll choose one unless there's a compelling reason not to do so. Were we to have two code paths—one just for old versions of IE—we'd actually need to run the tests in those old browsers. With one code path, running the tests in Node is sufficient.

Comment (buzzdecafe) [inline src/internal/_functionName.js]: in that case you have dragged all platforms down to IE's level. that hardly seems the preferable choice

Comment (davidchambers) [inline src/internal/_functionName.js]: > in that case you have dragged all platforms down to IE's level.

It sounds as though you *are* concerned about the performance implications. ;)

[MDN][1] states the following:

> This is a new technology, part of the ECMAScript 2015 (ES6) standard.

It seems the approach above is the "correct" approach, though it's undoubtedly ugly.


[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name

**Comment by davidchambers** [comment]
Is this approach reasonable, or do others share @buzzdecafe's concern?

**Pull request merged**